### ISSUE TITLE
💅 sibling-index is a real CSS 'function'

### ISSUE THREAD

**Issue by andre-lmarinho** [biomejs/biome#8920]
### Environment information

I don't really know what to say, sigling-index should not be a problem.

https://developer.mozilla.org/en-US/docs/Web/CSS/Reference/Values/sibling-index

Also tested with sibling-count, same error.

Termina error:

```BASH
src/utils/theme.css:79:27 lint/correctness/noUnknownFunction ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ Unexpected unknown function: sibling-index
  
    77 │     transform: translateY(var(--hero-reveal-from));
    78 │     animation: hero-reveal 800ms cubic-bezier(0.2, 0.8, 0.2, 1) forwards;
  > 79 │     animation-delay: calc(sibling-index() * 100ms);
       │                           ^^^^^^^^^^^^^
    80 │   }
    81 │ 
  
  ℹ Use a known function instead.

```

### Rule name

lint/correctness/noUnknownFunction

### Playground link

https://biomejs.dev/playground/?code=QABsAGEAeQBlAHIAIABjAG8AbQBwAG8AbgBlAG4AdABzACAAewAKACAAIAAuAGgAZQByAG8ALQBzAHQAYQBnAGcAZQByACAAPgAgACoAIAB7AAoAIAAgACAAIABvAHAAYQBjAGkAdAB5ADoAIAAwADsACgAgACAAIAAgAC0ALQBoAGUAcgBvAC0AcgBlAHYAZQBhAGwALQBmAHIAbwBtADoAIAAyADgAcAB4ADsACgAgACAAIAAgAHQAcgBhAG4AcwBmAG8AcgBtADoAIAB0AHIAYQBuAHMAbABhAHQAZQBZACgAdgBhAHIAKAAtAC0AaABlAHIAbwAtAHIAZQB2AGUAYQBsAC0AZgByAG8AbQApACkAOwAKACAAIAAgACAAYQBuAGkAbQBhAHQAaQBvAG4AOgAgAGgAZQByAG8ALQByAGUAdgBlAGEAbAAgADgAMAAwAG0AcwAgAGMAdQBiAGkAYwAtAGIAZQB6AGkAZQByACgAMAAuADIALAAgADAALgA4ACwAIAAwAC4AMgAsACAAMQApACAAZgBvAHIAdwBhAHIAZABzADsACgAgACAAIAAgAGEAbgBpAG0AYQB0AGkAbwBuAC0AZABlAGwAYQB5ADoAIABjAGEAbABjACgAcwBpAGIAbABpAG4AZwAtAGMAbwB1AG4AdAAoACkAIAAqACAAMQAwADAAbQBzACkAOwAKACAAIAB9AAoACgAgACAALgBoAGUAcgBvAC0AYQByAHIAbwB3AC0AcgBlAHYAZQBhAGwAIAB7AAoAIAAgACAAIABvAHAAYQBjAGkAdAB5ADoAIAAwADsACgAgACAAIAAgAC0ALQBoAGUAcgBvAC0AcgBlAHYAZQBhAGwALQBmAHIAbwBtADoAIAAtADEAMgBwAHgAOwAKACAAIAAgACAAdAByAGEAbgBzAGYAbwByAG0AOgAgAHQAcgBhAG4AcwBsAGEAdABlAFkAKAB2AGEAcgAoAC0ALQBoAGUAcgBvAC0AcgBlAHYAZQBhAGwALQBmAHIAbwBtACkAKQA7AAoAIAAgACAAIABhAG4AaQBtAGEAdABpAG8AbgA6ACAAaABlAHIAbwAtAHIAZQB2AGUAYQBsACAANgAwADAAbQBzACAAYwB1AGIAaQBjAC0AYgBlAHoAaQBlAHIAKAAwAC4AMgAsACAAMAAuADgALAAgADAALgAyACwAIAAxACkAIABmAG8AcgB3AGEAcgBkAHMAOwAKACAAIAAgACAAYQBuAGkAbQBhAHQAaQBvAG4ALQBkAGUAbABhAHkAOgAgADcAMAAwAG0AcwA7AAoAIAAgAH0ACgB9AAoACgBAAGsAZQB5AGYAcgBhAG0AZQBzACAAaABlAHIAbwAtAHIAZQB2AGUAYQBsACAAewAKACAAIABmAHIAbwBtACAAewAKACAAIAAgACAAbwBwAGEAYwBpAHQAeQA6ACAAMAA7AAoAIAAgACAAIAB0AHIAYQBuAHMAZgBvAHIAbQA6ACAAdAByAGEAbgBzAGwAYQB0AGUAWQAoAHYAYQByACgALQAtAGgAZQByAG8ALQByAGUAdgBlAGEAbAAtAGYAcgBvAG0ALAAgADAAcAB4ACkAKQA7AAoAIAAgAH0ACgAgACAAdABvACAAewAKACAAIAAgACAAbwBwAGEAYwBpAHQAeQA6ACAAMQA7AAoAIAAgACAAIAB0AHIAYQBuAHMAZgBvAHIAbQA6ACAAdAByAGEAbgBzAGwAYQB0AGUAWQAoADAAKQA7AAoAIAAgAH0ACgB9AAoACgBAAG0AZQBkAGkAYQAgACgAcAByAGUAZgBlAHIAcwAtAHIAZQBkAHUAYwBlAGQALQBtAG8AdABpAG8AbgA6ACAAcgBlAGQAdQBjAGUAKQAgAHsACgAgACAALgBoAGUAcgBvAC0AcwB0AGEAZwBnAGUAcgAgAD4AIAAqACwACgAgACAALgBoAGUAcgBvAC0AYQByAHIAbwB3AC0AcgBlAHYAZQBhAGwAIAB7AAoAIAAgACAAIABhAG4AaQBtAGEAdABpAG8AbgA6ACAAbgBvAG4AZQA7AAoAIAAgACAAIABvAHAAYQBjAGkAdAB5ADoAIAAxADsACgAgACAAIAAgAHQAcgBhAG4AcwBmAG8AcgBtADoAIABuAG8AbgBlADsACgAgACAAfQAKAH0A&language=css

### Expected result

sibling-index and sibling-count is a real thing on CSS.

Maybe we can flag as a soft warning because dont have 100% Browser compatibility? I don't know. Don't think that this is useful but...

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by ematipico** [comment]
Wanna send a PR @andre-lmarinho ? It's a real easy fix, and a great opportunity to contribute

**Comment by andre-lmarinho** [comment]
Hi @ematipico!  Thanks for the nudge. Yes, I’d be happy to open a PR.

I originally reported the issue around the CSS functions, and I was already thinking about doing a broader review, especially with how much has been added to the CSS layer recently. I’ll take a look and follow up with a PR.
